assert/assert.go

```diff
@@ -100,13 +100,15 @@ import (
 
 // BoolOrComparison can be a bool, cmp.Comparison, or error. See Assert for
 // details about how this type is used.
-type BoolOrComparison interface{}
+type BoolOrComparison interface {
+	bool | func() (bool, string) | ~func() cmp.Result
```

Comment:
This might need to include `error`. I believe it was previously possible to use `assert.Assert(t, err)`.